std/range/package.d

```diff
@@ -4355,15 +4357,348 @@ if (Ranges.length && allSatisfy!(isInputRange, Ranges))
     }
 }
 
+/+
+Non-public. Like $(LREF Zip) with `StoppingPolicy.shortest`
+except it properly implements `back` and `popBack` in the
+case of uneven ranges or disables those operations when
+it is not possible to guarantee they are correct.
++/
+package template ZipShortest(Ranges...)
+if (Ranges.length && __traits(compiles,
```

Comment:
`if (Ranges.length && allSatisfy!(isInputRange, Ranges))`?

Reply:
It needs to be written like that to coexist with `ZipShortest(Flag!"allKnownSameLength" allKnownSameLength, Ranges...)`.